### PR TITLE
Switch EC2 example config to use AWS deep learning AMI + latest Ray wheel

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

This is the fix-up PR for https://github.com/ray-project/ray/pull/1311

### CODE REVIEW

File: python/ray/autoscaler/aws/config.py [hunk 1]
```diff
@@ -138,6 +138,11 @@ def _configure_subnet(config):
         [s for s in ec2.subnets.all() if s.state == "available"],
         reverse=True,  # sort from Z-A
         key=lambda subnet: subnet.availability_zone)
+    if not subnets:
+        raise Exception(
+            "No subnets found, try manually creating an instance in "
+            "your specified region to populate the list of subnets "
+            "and trying this again.")
```
Comment (ericl) [inline python/ray/autoscaler/aws/config.py]: I'm not sure why the subnets list is empty until you do this; I ran into this problem when trying to use the script in a region I hadn't used before (us-west-2).

File: python/ray/autoscaler/aws/example.yaml [hunk 2]
```diff
@@ -58,15 +58,13 @@ file_mounts: {
 
 # List of shell commands to run to initialize the head node.
 head_init_commands:
-    - cd ~/ray; git remote add eric https://github.com/ericl/ray.git || true
-    - cd ~/ray; git fetch eric && git reset --hard e1e97b3
+    - sudo pip3 install -U https://s3-us-west-2.amazonaws.com/ray-wheels/f5ea44338eca392df3a868035df3901829cc2ca1/ray-0.3.0-cp35-cp35m-manylinux1_x86_64.whl
```
Comment (robertnishihara) [inline python/ray/autoscaler/aws/example.yaml]: Would it work if we get rid of `sudo` and add in `--user`

**Comment by robertnishihara** [comment]
Thanks @ericl this looks good.

Separately, what is the best way for developers to use this? E.g., should developers make a new AMI  with Ray installed via `python setup.py develop` and then change the config to use that AMI?

Also, for development, would it make sense to pre-populate a `workers.txt` file on the head node? And the `start_worker.sh`/`stop_worker.sh`/`upgrade.sh` scripts?

**Review by robertnishihara — APPROVED** [review]
I left one comment. Feel free to address it or merge anyway.

**Comment by ericl** [comment]
@robertnishihara made that change, it seems to work.

> Separately, what is the best way for developers to use this? E.g., should developers make a new AMI with Ray installed via python setup.py develop and then change the config to use that AMI?

Yeah, I snapshotted a new AMI that had my repo pre-cloned.

> Also, for development, would it make sense to pre-populate a workers.txt file on the head node? And the start_worker.sh/stop_worker.sh/upgrade.sh scripts?

What I've been doing is adding a `git checkout <my_sha>` in the init commands and then running `ray create_or_update`, which updates the nodes. When I need to force restart the nodes I add some dummy init command (e.g. "echo foo") that forces a update of all nodes.

**Comment by robertnishihara** [comment]
It's got to be unrelated to this PR, but this is the first time I saw the error

```
======================================================================
FAIL: testCleanupOnDriverExitManyRedisShards (__main__.MonitorTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/monitor_test.py", line 88, in testCleanupOnDriverExitManyRedisShards
    self._testCleanupOnDriverExit(num_redis_shards=5)
  File "test/monitor_test.py", line 79, in _testCleanupOnDriverExit
    self.assertEqual((0, 1), StateSummary()[:2])
AssertionError: Tuples differ: (0, 1) != (4, 3)

First differing element 0:
0
4

- (0, 1)
+ (4, 3)
```

cc @concretevitamin

**Comment by robertnishihara** [comment]
Filed an issue at #1332.

**Pull request merged**